### ISSUE TITLE
Lint rule for missing yield statement

### ISSUE THREAD

**Issue by RShpd** [passsy/dart-lint#24]
In my bloc I have following code:
```
  @override
  Stream<VerificationState> mapEventToState(VerificationEvent event) async* {
    yield* event.when(
      loginRetry: (phoneNumber) => _loginRetry( phoneNumber ),
      verify: (phoneNumber, code) => _verify(phoneNumber, code),
    );
  }
```
If I forget to add `yield*`, lint does not warn me. How can I be warned for such a mistake?

**Comment by passsy** [comment]
static analysis can't help here. A `async*` function without `yield`/`yield*` is fine, returning a `Stream` without elements. In many cases that's exactly what you want.

**Comment by RShpd** [comment]
Then can't I add a custom rule that warn me  about this?
